src/lambda/wizards/samDeployWizard.ts

```diff
@@ -821,3 +882,22 @@ async function populateS3QuickPick(
         }
     })
 }
+
+/**
+ * Creates a new S3 bucket and returns its name
+ * @param name What the new bucket will be named
+ * @param selectedRegion Region for the S3 client
+ * @returns If successful, returns the new name of the bucket created, undefined if not
+ */
+async function createNewS3Bucket(name: string, selectedRegion: string): Promise<string | undefined> {
```

Comment:
Any reason why this is here and not in the `DefaultS3Client`?

Reply:
Most of it already is, deleted this function and moved logic to the `S3_BUCKET` step.